### DISCUSSION TITLE
Can the Purchases view show share of purchase count in addition to revenue?

### DISCUSSION THREAD

**Discussion by mattcorey** [TelemetryDeck/Viewer#447] [Ideas]
Just looked at my Purchases dashboard (the built in one), and was surprised to see that I have a shocking large percentage of revenue in the last month from macOS -- I don't directly support macOS, so this would be by way of the iPad app running on Apple Silicon, which was very surprising.

After more digging, it turns out that I had a few lifetime purchases made on the platform, which are much higher than my subscription rates, and therefore throws off that chart.  Having the same charts show percentage of individual purchases would be helpful to balance this out - do you have the data necessary to build these charts automatically?

**Comment by winsmith** [comment]
Paging @Jeehut – is this something that's easily feasible?

**Comment by Jeehut** [comment]
Just for reference, I'm adding a screenshot of the built-in insights with some sample data:
<img width="1341" alt="Bildschirmfoto 2024-09-17 um 18 35 22" src="https://github.com/user-attachments/assets/2178246f-0e59-42eb-a974-d3aa817768d4">

@mattcorey  To make sure I'm not misunderstanding you: You are looking at the "Revenue by Platform (in USD)" insight and that is the chart "thrown off" due to lifetime purchases?

Also, you said "after more digging" you noticed that it was due to lifetime purchases. Didn't the "Revenue by Type (in USD)" chart on the top right already communicate that fact (so no need for extra digging)? Just asking so I can make sure to fix any issues with that in case it didn't work correctly for you.

Assuming I understood both correctly, what you want is that we introduce charts that only show stats for subscriptions only, right? There are various ways we could do that. The simplest would be to add an extra insight for each of Purchase History + Revenue by Platform/Currency/Country that is "subscriptions only" and also for "in-app purchases only". But that would also bloat the screen quite a bit. Another solution could be that we add a Filter to the top of that screen that would apply to all insights and remove the "Revenue by Type" insight (as we would filter by that).

That kind of solution would be "easily feasible". (@winsmith) The relevant code that makes this possible is [this line](https://github.com/TelemetryDeck/SwiftSDK/blob/9aeb5d74b306c8ab983149a0703ce926142fa0ba/Sources/TelemetryClient/Presets/TelemetryDeck%2BPurchases.swift#L62). If what you want is to filter by a specific purchase (such as "Premium Monthly"), we would need to additionally send the product ID of the purchase (which we currently don't do). Then we could even add a filter picker for the different products you offer.

Please let me know if I misunderstood anything and let me know, what aspect has the biggest priority for you.

**Reply by mattcorey** [reply]
Hi @Jeehut,

In the case of your sample data, the 'Revenue by Type' chart would, in fact, show that there was an unusually high rate of one-time purchases vs. subscriptions, however this sample data isn't reflective of what I saw at the time - if memory serves, I had a roughly 60/40 split between iOS and macOS, with the majority of macOS purchases being the higher rate lifetime option, while iOS had nearly zero lifetime purchases.  Because of this, I was unable to correlate the data in the 'Revenue by Type' chart to what I was seeing in the Platform chart.  (I'm currently about 85/15%, so the story has changed a bit as of today).

What my original ask was for was a simple "Purchases by Platform" view, which would show raw counts - this, combined with "Revenue by Platform" would have very quickly shown me that I only had a handful of macOS purchases, but they produce disproportionately high percentage of revenue.  I think having a "Purchases by [group by category]" for each of these would quickly disambiguate any anomalies we see in raw revenue data.

What would be even more interesting would be the ability to click on the Mac segment of the 'Revenue by Platform' chart, and have that filter all of the other charts, so I could immediately see the breakdowns of Type, Country, Currency, etc.  (I recognize this is a much more complicated ask 😆, and really, what I describe above would have identified the anomaly I saw much more quickly)

**Reply by Jeehut** [reply]
Thank you for the thorough explanation. I think I understand your problem better now. Having that said, I don't quite get what you mean by "Purchases by [group by category]"? Do you mean "Purchases by Type/Platform/Country/Currency" and for each of these, instead of using the sum on the revenue, you would expect simply the count of purchases? Did I get that right?
